lib/shrine.rb

```diff
@@ -600,14 +600,14 @@ def swap(uploaded_file)
         # Deletes the previous attachment that was replaced, typically called
         # after the model instance is saved with the new attachment.
         def replace
-          _delete(@old, action: :replace) if @old && !cache.uploaded?(@old)
+          _delete(@old, action: :replace) if @old && !cached?(@old)
         end
 
         # Deletes the current attachment, typically called after destroying the
         # record.
         def destroy
           file = get
-          _delete(file, action: :destroy) if file && !cache.uploaded?(file)
+          _delete(file, action: :destroy) if file && !cached?(file)
```

Comment:
can be simplified to `... if !cached?(file)` since `file` is not `nil` is checked in `#cached?`.

Reply:
I believe you are incorrect. 

```
nil && !cached?(nil) #=> nil
!cached?(nil) #=> true
```

I believe the nil check is needed or it will have different and wrong semantics.

Reply:
`cached?(file)` resolves to `file && cache.uploaded?(file)`. If you say `!cached?`, that resolves to `!file || !cache.uploaded?(file)`, which is different than `file && !cache.uploaded?(file)`. Particularly, here it would result in deletion being called if the file was `nil`.

Reply:
@jrochkind Oops, you're right! I don't like the duplicate `nil` check but whatever... not the most important thing.

Reply:
hey, a nil check is probably the cheapest possible thing you can do in ruby, no big deal.